src/rust/engine/src/externs/interface.rs

```diff
@@ -1647,6 +1647,7 @@ fn write_digest(
   digest: &PyAny,
   path_prefix: String,
   clear_paths: Vec<String>,
+  invalidate: bool,
```

Comment:
Potentially this is unnecessary and this should unconditionally invalidate?

Reply:
Yes, I believe that it should unconditionally invalidate.